### PR TITLE
[Test] Add test suite for NSA backend

### PR DESCRIPTION
**Pull request by Johnsonms**
## Motivation
https://github.com/sgl-project/sglang/issues/14523

## Modifications
  20+ test cases covering:

  1. Core forward operations
    - test_forward_extend - Standard extend operation
    - test_forward_decode - Decode with cached tokens
    - test_forward_extend_with_prefix - Extending from cached prefix
  2. Metadata structure validation
    - NSA metadata creation and structure for decode/extend modes
    - Page table shape and correctness
    - Cumulative sequence lengths computation
    - NSA-specific seqlens clipping to index_topk
  3. Edge cases
    - Single token decode
    - Extend with no prefix (cold start)
    - Extend with full prefix
    - Variable batch sizes (1, 3, 8)
    - Variable sequence lengths (1, 64, 256, 512)
  4. Data consistency
    - Dtype consistency (int32 for all seqlens and page tables)
    - Device consistency (all tensors on CUDA)
    - NSA cumulative seqlens consistency
    - nsa_seqlens_expanded shape and monotonicity

## Accuracy Tests
`python python/sglang/test/attention/test_nsa_backend.py`
image

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).
- [x] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

### CODE REVIEW

File: python/sglang/test/attention/test_nsa_backend.py [hunk 1]
```diff
@@ -0,0 +1,709 @@
...
+                ),
+                seq_lens_cpu=torch.tensor([total_len] * self.batch_size, device="cpu"),
+                attn_backend=self.backend,
+            )
+
+        # Add token pool from model runner to forward batch
+        forward_batch.req_to_token_pool = self.model_runner.req_to_token_pool
+        forward_batch.token_to_kv_pool = self.model_runner.token_to_kv_pool
+
+        return forward_batch
```
[the remark this answers is not in the document]
Comment (Johnsonms) [reply python/sglang/test/attention/test_nsa_backend.py:269]: My code is more readable, easy to understood
